src/activity_handlers.cpp

```diff
@@ -242,7 +255,47 @@ void set_up_butchery( player_activity &act, player &u )
         return;
     }
 
+    bool has_table_nearby = false;
+    for( const tripoint &pt : g->m.points_in_radius( u.pos(), 1 ) ) {
+        if( g->m.furn( pt ) == furn_id( "f_table" ) ) {
```

Comment:
Don't hardcode this to table only. Add a flag like `TABLE` to the flat surface furniture and check for it here instead.